client/rpc/block.go

```diff
@@ -62,26 +67,59 @@ func getBlock(clientCtx client.Context, height *int64) ([]byte, error) {
 	// header -> BlockchainInfo
 	// header, tx -> Block
 	// results -> BlockResults
-	res, err := node.Block(context.Background(), height)
+	resBlock, err := node.Block(context.Background(), height)
 	if err != nil {
 		return nil, err
 	}
 
-	return legacy.Cdc.MarshalJSON(res)
+	out, err := mkBlockResult(resBlock)
+	if err != nil {
+		return out, err
+	}
+
+	return out, nil
 }
 
-// get the current blockchain height
-func GetChainHeight(clientCtx client.Context) (int64, error) {
+func GetBlockByHash(clientCtx client.Context, hashHexString string) (*tm.Block, error) {
+	hash, err := hex.DecodeString(hashHexString)
+	if err != nil {
+		return nil, err
+	}
+
+	// get the node
 	node, err := clientCtx.GetNode()
 	if err != nil {
-		return -1, err
+		return nil, err
 	}
 
-	status, err := node.Status(context.Background())
+	resBlock, err := node.BlockByHash(context.Background(), hash)
+
+	if err != nil || resBlock.Block == nil {
+		return nil, fmt.Errorf("block not found with Hash: %s with Error: %s", hashHexString, err)
```

Comment:
```suggestion
		return nil, fmt.Errorf("block not found with hash: %s with error: %s", hashHexString, err)
```

Could we maybe have two different errors? Otherwise, when `resBlock.Block` is nil, then the error may or may not be nil.
Like:

```sh
Error: block not found with Hash: 7E193092316E69BBF8ACB81E34F9831DF3BBD9C2C276DE520F6B2828C3FB9F05 with Error: %!s(<nil>)
```

Reply:
What I mean here, was to have two different if statements